nav2_smac_planner/src/node_hybrid.cpp

```diff
@@ -402,72 +415,89 @@ void NodeHybrid::resetObstacleHeuristic(
       obstacle_heuristic_lookup_table.begin(), obstacle_size, 0.0);
   }
 
+  obstacle_heuristic_queue.clear();
+  obstacle_heuristic_queue.reserve(
+    sampled_costmap->getSizeInCellsX() * sampled_costmap->getSizeInCellsY());
+
   // Set initial goal point to queue from. Divided by 2 due to downsampled costmap.
-  std::queue<unsigned int> q;
-  std::swap(obstacle_heuristic_queue, q);
-  obstacle_heuristic_queue.emplace(
-    ceil(goal_y / 2.0) * sampled_costmap->getSizeInCellsX() + ceil(goal_x / 2.0));
+  const unsigned int size_x = sampled_costmap->getSizeInCellsX();
+  const unsigned int goal_index = floor(goal_y / 2.0) * size_x + floor(goal_x / 2.0);
```

Comment:
You changed `ceil` to `floor` here, was that intentional?

Reply:
As well below in L440-441, was there something wrong with the current behavior?

Reply:
Yes, I think that `ceil` was there for some mysterious historical reasons or something, because it is just wrong and points to an incorrect cell. Nodes use [`static_cast<unsigned int>`](https://github.com/ros-planning/navigation2/blob/85cf934de1c0cf8d335f8d46440a96636410b8ee/nav2_smac_planner/src/collision_checker.cpp#L97) which is equivalent of `floor` for positive numbers and downsampler uses [`coord*2 + (0 or 1)`](https://github.com/ros-planning/navigation2/blob/85cf934de1c0cf8d335f8d46440a96636410b8ee/nav2_smac_planner/src/costmap_downsampler.cpp#L136) which is again inverse of `floor`, not `ceil`. Also verified by logs during analysis and debugging.

Reply:
OK